example/panel/modules/sample_tracer.dart

```diff
@@ -0,0 +1,162 @@
+import 'dart:async';
+import 'package:opentracing/opentracing.dart';
+
+class SampleSpan implements Span {
+  static int _nextId = 0;
+  final int _id = _nextId++;
+
+  @override
+  final List<Reference> references;
+
+  @override
+  final Map<String, dynamic> tags;
+
+  @override
+  final List<LogData> logData = [];
+
+  @override
+  final String operationName;
+
+  @override
+  SpanContext context;
+
+  @override
+  final DateTime startTime;
+  DateTime _endTime;
+
+  Completer<Span> _whenFinished = new Completer<Span>();
+
+  SampleSpan(
+    this.operationName, {
+    SpanContext childOf,
+    this.references,
+    DateTime startTime,
+    Map<String, dynamic> tags,
+  })
+      : this.startTime = startTime ?? new DateTime.now(),
+        this.tags = tags ?? {} {
+    if (childOf != null) {
+      references.add(new Reference.childOf(childOf));
+    }
+    setTag('span.kind', 'client');
+
+    final parent = parentContext;
+    if (parent != null) {
+      this.context = new SpanContext(spanId: _id, traceId: parent.traceId);
+      this.context.baggage.addAll(parent.baggage);
+    } else {
+      this.context = new SpanContext(spanId: _id, traceId: _id);
+    }
+  }
+
+  @override
+  void addTags(Map<String, dynamic> newTags) => tags.addAll(newTags);
+
+  @override
+  Duration get duration => _endTime?.difference(startTime);
```

Comment:
❓Are consumers of `Span` aware that this may be null?

Reply:
I figure it's not important here because it's just in the examples.